pypy3 regression: tests/test_api_jws.py::TestJWS::test_options_must_be_dict - ValueError: sequence of pairs expected

The 2.0.0 release introduces a regression in PyPy 3 support.

## Expected Result

Tests passing with pypy3.

## Actual Result

```pytb
==================================================================== FAILURES =====================================================================
________________________________________________________ TestJWS.test_options_must_be_dict ________________________________________________________

self = <tests.test_api_jws.TestJWS object at 0x00007f705bc57670>, jws = <jwt.api_jws.PyJWS object at 0x00007f705bc46020>

    def test_options_must_be_dict(self, jws):
        pytest.raises(TypeError, PyJWS, options=object())
>       pytest.raises(TypeError, PyJWS, options=("something"))

tests/test_api_jws.py:86: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jwt.api_jws.PyJWS object at 0x00007f705b5a26e8>, algorithms = None, options = 'something'

    def __init__(self, algorithms=None, options=None):
        self._algorithms = get_default_algorithms()
        self._valid_algs = (
            set(algorithms) if algorithms is not None else set(self._algorithms)
        )
    
        # Remove algorithms that aren't on the whitelist
        for key in list(self._algorithms.keys()):
            if key not in self._valid_algs:
                del self._algorithms[key]
    
        if options is None:
            options = {}
>       self.options = {**self._get_default_options(), **options}
E       ValueError: sequence of pairs expected

jwt/api_jws.py:37: ValueError
____________________________________________________ TestJWS.test_decode_options_must_be_dict _____________________________________________________

self = <tests.test_api_jws.TestJWS object at 0x00007f7059ca8bb8>, jws = <jwt.api_jws.PyJWS object at 0x00007f7059ca8758>, payload = b'hello world'

    def test_decode_options_must_be_dict(self, jws, payload):
        token = jws.encode(payload, "secret")
    
        with pytest.raises(TypeError):
            jws.decode(token, "secret", options=object())
    
        with pytest.raises(TypeError):
>           jws.decode(token, "secret", options="something")

tests/test_api_jws.py:582: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
jwt/api_jws.py:165: in decode
    decoded = self.decode_complete(jwt, key, algorithms, options, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jwt.api_jws.PyJWS object at 0x00007f7059ca8758>
jwt = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.aGVsbG8gd29ybGQ.gEW0pdU4kxPthjtehYdhxB9mMOGajt1xCKlGGXDJ8PM', key = 'secret', algorithms = None
options = 'something', kwargs = {}

    def decode_complete(
        self,
        jwt: str,
        key: str = "",
        algorithms: List[str] = None,
        options: Dict = None,
        **kwargs,
    ) -> Dict[str, Any]:
        if options is None:
            options = {}
>       merged_options = {**self.options, **options}
E       ValueError: sequence of pairs expected

jwt/api_jws.py:138: ValueError
```

## Reproduction Steps

```sh
tox -e pypy3-crypto
```

## System Information

    $ pypy3 -m jwt.help


```
{
  "cryptography": {
    "version": "3.3.1"
  },
  "implementation": {
    "name": "PyPy",
    "version": "7.3.3beta"
  },
  "platform": {
    "release": "5.10.1",
    "system": "Linux"
  },
  "pyjwt": {
    "version": "2.0.0"
  }
}
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -18,7 +18,8 @@ jobs:
     strategy:
       matrix:
         platform: ["ubuntu-latest", "windows-latest"]
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10",
+                         "3.11.0-beta - 3.11", "pypy-3.8", "pypy-3.9"]
 
     steps:
       - uses: "actions/checkout@v3"

--- a/tests/test_api_jws.py
+++ b/tests/test_api_jws.py
@@ -83,7 +83,7 @@ class TestJWS:
 
     def test_options_must_be_dict(self, jws):
         pytest.raises(TypeError, PyJWS, options=object())
-        pytest.raises(TypeError, PyJWS, options=("something"))
+        pytest.raises((TypeError, ValueError), PyJWS, options=("something"))
 
     def test_encode_decode(self, jws, payload):
         secret = "secret"
@@ -607,7 +607,7 @@ class TestJWS:
         with pytest.raises(TypeError):
             jws.decode(token, "secret", options=object())
 
-        with pytest.raises(TypeError):
+        with pytest.raises((TypeError, ValueError)):
             jws.decode(token, "secret", options="something")
 
     def test_custom_json_encoder(self, jws, payload):

--- a/tox.ini
+++ b/tox.ini
@@ -13,13 +13,17 @@ python =
     3.8: py38, typing
     3.9: py39
     3.10: py310
+    3.11: py311
+    pypy-3.7: pypy3
+    pypy-3.8: pypy3
+    pypy-3.9: pypy3
 
 
 [tox]
 envlist =
     lint
     typing
-    py{36,37,38,39}-{crypto,nocrypto}
+    py{36,37,38,39,310,311,py3}-{crypto,nocrypto}
     docs
     pypi-description
     coverage-report

--- a/tox.ini
+++ b/tox.ini
@@ -14,7 +14,6 @@ python =
     3.9: py39
     3.10: py310
     3.11: py311
-    pypy-3.7: pypy3
     pypy-3.8: pypy3
     pypy-3.9: pypy3
 

